File: hello/C-OMP/hello.c
#include <stdio.h>
#include <omp.h>

int main(void)
{
  #pragma omp parallel
  {
    #pragma omp master
    {
      printf("Running on %d threads\n", omp_get_num_threads());
    }

    printf("Hello from thread %d\n", omp_get_thread_num());
  }

  return 0;
}
  

File: hello/F-OMP/hello.f90
program hello

  use omp_lib

  implicit none

  !$omp parallel

  !$omp master
  write(*,*) "Running on ", omp_get_num_threads(), " threads"
  !$omp end master

  write(*,*) "Hello from thread ", omp_get_thread_num()

  !$omp end parallel

end program hello

